lib/absinthe/lexer.ex

```diff
@@ -242,12 +242,88 @@ defmodule Absinthe.Lexer do
     end
   end
 
-  defp convert_token_column({ident, loc, data}, lines) do
-    {ident, byte_loc_to_char_loc(loc, lines), data}
+  defp convert_token_columns_from_byte_to_char(tokens, [first_line | next_lines]) do
+    initial_cursor_state = %{
+      line_num_cursor: 1,
+      current_line_substring: first_line,
+      current_line_char_offset: 1,
+      current_line_byte_offset: 1,
+      next_lines: next_lines
+    }
+
+    Enum.reduce(tokens, {[], initial_cursor_state}, fn current_token, {results, cursor_state} ->
```

Comment:
Wow, really nice improvement! 

One tiny nitpick is to use Enum.map_reduce here. Cuts down on the boilerplate of Enum.reduce a little.

Reply:
Oh! Great suggestion, thanks @maartenvanvliet ! Today I learned about `Enum.map_reduce`...

 I'll update the PR.